comp/core/workloadmeta/collectors/internal/process/process_collector.go

```diff
@@ -416,12 +418,14 @@ func (c *collector) getProcessEntitiesFromServices(newPids []int32, heartbeatPid
 	entities := make([]*workloadmeta.Process, 0, len(pidsToService))
 	now := c.clock.Now().UTC()
 
+
 	// Process new PIDs - create complete entities
 	for _, pid := range newPids {
 		service := pidsToService[pid]
 		if service == nil {
 			// Handle injected processes that are not services
 			if injectedPids.Has(pid) {
+				c.injectedOnlyPids.Add(pid) // Track for cleanup
```

Comment:
Should the `if` condition also ignore PIDs which are already in `injectedOnlyPids`?  Otherwise I think a redundant entity will be added for every retries (probably doesn't have any real effect though).